avaframe/ana3AIMEC/ana3AIMEC.py

```diff
@@ -540,347 +393,140 @@ def assignData(fnames, rasterIndData):
-                      facecolor=[.8, .8, .8], alpha=0.5)
-    ref1 = mpatches.Patch(alpha=0.5, color=[.8, .8, .8])
-    ax2.plot(p_median, s_coordinate, color='r')
-    ref2 = mlines.Line2D([], [], color='r', linewidth=2)
-#    ax2.plot(p_mean, s_coordinate, color='b')
-#    ref3 = mlines.Line2D([], [], color='b', linewidth=2)
-    ax2.set_ylabel('s [m]')
-    ax2.set_ylim([yy.min(), yy.max()])
-    ax2.set_xlim(auto=True)
-    ax2.set_xlabel('Pmax(s) [kPa]')
-    ax2.legend((ref1, ref2),
-               ('quantiles', 'median'), loc=0)
-
-    fig.tight_layout()
-
-    if cfgFlags.getboolean('savePlot'):
-        pro_name = fname[0].split('/')[-3]
-        outname_fin = ''.join([outpath, '/pics/', pro_name, '_dptr',
-                               str(int(p_lim)), '_simulationsl', '.pdf'])
-        if not os.path.exists(os.path.dirname(outname_fin)):
-            os.makedirs(os.path.dirname(outname_fin))
-        fig.savefig(outname_fin, transparent=True)
-
-    if cfgFlags.getboolean('plotFigure'):
-        plt.show()
-    else:
-        plt.ioff()
-
-    return runout, runout_mean, ampp, mmpp, amd, mmd
+        elevRel[i] = dataDEM[cupper, int(np.floor(n/2)+1)]
+        deltaH[i] = dataDEM[cupper, int(np.floor(n/2)+1)] - dataDEM[clower, int(np.floor(n/2)+1)]
+
+        # analyze mass
+        try:
+            releaseMass[i], entrainedMass[i], grIndex[i], grGrad[i] = read_write(fname_mass[i])
+            if (releaseMass == releaseMass[0]).any():
+                releaseMass = releaseMass[0]
+            else:
+                log.warning('Release masses differs between simulations!')
+        except IndexError:
+            releaseMass[i] = np.NaN
+            entrainedMass[i] = np.NaN
+            grIndex[i] = np.NaN
+            grGrad[i] = np.NaN
+
+        # log.info('%s\t%10.4f\t%10.4f\t%10.4f' % (i+1, runout[i], ampp[i], amd[i]))
+        log.info('{: <15} {:<15.4f} {:<15.4f} {:<15.4f}'.format(*[i+1, runout[i], ampp[i], amd[i]]))
+
+    resAnalysis['runout'] = runout
+    resAnalysis['runout_mean'] = runout_mean
+    resAnalysis['AMPP'] = ampp
+    resAnalysis['MMPP'] = mmpp
+    resAnalysis['AMD'] = amd
+    resAnalysis['MMD'] = mmd
+    resAnalysis['elevRel'] = elevRel
+    resAnalysis['deltaH'] = deltaH
+    resAnalysis['relMass'] = releaseMass
+    resAnalysis['entMass'] = entrainedMass
+    resAnalysis['growthIndex'] = grIndex
+    resAnalysis['growthGrad'] = grGrad
+
+    # prepare for plot
+    inputPlot = {}
+    inputPlot['p_mean'] = p_cross_all.mean(axis=0)
+    inputPlot['p_median'] = np.median(p_cross_all, axis=0)
+    inputPlot['p_percentile'] = sp.percentile(p_cross_all, [2.5, 50, 97.5], axis=0)
+    inputPlot['dataPressure'] = newRasters['newRasterPressure']
+    inputPlot['runout'] = resAnalysis['runout']
+    inputPlot['runout_mean'] = resAnalysis['runout_mean']
+    inputPlot['pressureLimit'] = p_lim
+
+    outAimec.visu_runout(raster_transfo, inputPlot, cfgPath, cfgFlags)
```

Comment:
Should this be in a plotFlag? In case someone doesn't want to have plots? Or do we deem plots essential for aimec results?

Reply:
this function depending on the flags, plots and or saves the plots. If both flags are set to False nothing happens